crates/mpz-common/src/executor/mt.rs

```
@@ -230,6 +273,66 @@ where
         &mut self.io
     }
 
+    async fn map<'a, F, T, R, W>(
+        &'a mut self,
+        items: Vec<T>,
+        f: F,
+        weight: W,
+    ) -> Result<Vec<R>, ContextError>
+    where
+        F: for<'b> Fn(&'b mut Self, T) -> ScopedBoxFuture<'static, 'b, R> + Clone + Send + 'static,
+        T: Send + 'static,
+        R: Send + 'static,
+        W: Fn(&T) -> usize + Send + 'static,
+    {
+        let item_count = items.len();
```

Do we want to return an early error here if `item_count` == 0 ?

---

doesnt seem necessary